beacon-chain/db/filesystem/ephemeral.go

```diff
@@ -0,0 +1,17 @@
+package filesystem
+
+import "github.com/spf13/afero"
+
+// NewEphemeralBlobStorage should only be used for tests.
+// The instance of BlobStorage returned is backed by an in-memory virtual filesystem,
+// improving test performance and simplifying cleanup.
+func NewEphemeralBlobStorage() *BlobStorage {
+	return &BlobStorage{fs: afero.NewMemMapFs()}
+}
+
+// NewEphemeralBlobStorageWithFs can be used by tests that want access to the virtual filesystem
+// in order to interact with it outside the parameters of the BlobStorage api.
+func NewEphemeralBlobStorageWithFs() (afero.Fs, *BlobStorage) {
+	fs := afero.NewMemMapFs()
+	return fs, &BlobStorage{fs: fs}
+}
```

Comment:
Is there a tag we can add to enforce this is test only and fail if it's used otherwise of test scope?

Reply:
In pure go a `_test.go` file can export symbols to other test packages, but I think we still haven't figured out a good way to do that with Bazel cc @prestonvanloon 

Cheap trick to enforce this: `NewEphemeralBlobStorage(_ *testing.T)`.

Reply:
https://github.com/prysmaticlabs/prysm/pull/13190/commits/977e5bafc7830fb9d6b30949db32cbe315bc8a33